gt/main.c

```diff
@@ -548,7 +552,8 @@ gt_neigh_get_ether_cache(struct neighbor_hash_table *neigh,
 			return NULL;
 		}
 	} else if (likely(inner_ip_ver == ETHER_TYPE_IPv6)) {
-		if (!lls_conf->nd_cache.ip_in_subnet(iface, ip_dst)) {
+		memcpy(&addr.ip.v6, ip_dst, sizeof(addr.ip.v6));
+		if (!lls_conf->nd_cache.ip_in_subnet(iface, &addr)) {
```

Comment:
As ``ipv6_in_subnet()`` only uses the field ``ip.v6`` in ``struct ipaddr``, maybe just using ``const void *addr`` as the parameter of ``ip_in_subnet()``, which points to the actual 16-byte IPv6 address, will be simpler and saves the function call ``memcpy()``?